sched/environ/env_unsetenv.c

```diff
@@ -61,23 +61,22 @@ int unsetenv(FAR const char *name)
 {
   FAR struct tcb_s *rtcb = this_task();
   FAR struct task_group_s *group = rtcb->group;
-  int ret = OK;
+  int idx;
 
   DEBUGASSERT(name && group);
 
   /* Check if the variable exists */
 
   sched_lock();
-  if (group && (ret = env_findvar(group, name)) >= 0)
+  if (group && (idx = env_findvar(group, name)) >= 0)
     {
       /* It does!  Remove the name=value pair from the environment. */
 
-      env_removevar(group, ret);
-      ret = OK;
+      env_removevar(group, idx);
     }
 
   sched_unlock();
-  return ret;
+  return OK;
```

Comment:
According to POSIX `unsetenv` can fail with
```
[EINVAL]
    The name argument points to an empty string, or points to a string containing an '=' character. 
```
Leaving here just for a note. Should be fixed by a separate PR.

Reply:
Yes we could add check for the incoming parameter too. The man page is a bit ambiguous in my opinion about this.

For `setenv()`it makes perfect sense to check for the incoming parameter. 
For `unsetenv()` it is ambiguous. The check is for parameter validity, but no such env variable can exist, as setenv() checks for parameter validity. As such name does not exist, should the function return success ?

Reply:
Yup, there has clearly been some iteration with that since I can find a contradicting (at least ambiguous!) man page (first result in google): https://man7.org/linux/man-pages/man3/setenv.3.html

But like said, neither **prohibit** returning EINVAL if the incoming parameter "name" is not valid so it is better to do the test.

Reply:
Added the parameter check as separate commit